Bottom: 0f702ea54455bf2b9f613ce96c8cc1126ef18e24
Top:    0bbde8afed5c0d53a5652918c923fca123bb5ea1
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-03-23 00:10:53 +0000

Fix to ParseGroupString()


---

diff --git a/src/misc.c b/src/misc.c
index 4f381ae..1146eba 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -1101,7 +1101,14 @@ ParseGroupString (char *group_string, LayerGroupType *LayerGroup, int *LayerN)
           /* ignore white spaces and check for separator */
           while (*s && isspace ((int) *s))
             s++;
-          if (!*s || *s == ':')
+
+          /* Backup so the outer loop terminates on the NULL termination */
+          if (*s == '\0') {
+            s--;
+            break;
+          }
+
+          if (*s == ':')
             break;
         }
       LayerGroup->Number[group] = member;
